### PR TITLE
Support OCI media types on pull in addition to docker media types

### LINKED ISSUE #74: oci_pull: handle OCI-format images
We should be able to pull an image that's published in OCI format, for example `crane manifest quay.io/buildah/stable`

Also it means we should handle more Media Types:
```
application/vnd.oci.image.index.v1+json
application/vnd.oci.image.manifest.v1+json
```

### PR DESCRIPTION
**Pull request by bcmyers**
This PR is composed of two commits:

1. Fixes a bug in `rules_pkg`'s sha256

2. Fixes #74. Adds support for OCI media types. Currently, `pull` only supports the media types from the docker standard, but it should also support the media types from the OCI standard.

Relevant specs:
* [Docker standard](https://github.com/distribution/distribution/blob/main/docs/spec/manifest-v2-2.md#media-types
) (already supported)
* [OCI standard](https://github.com/opencontainers/image-spec/blob/main/media-types.md) (support added in this PR)

Error this PR is fixing:

Setup:
```
bazel_dep(name = "aspect_bazel_lib", version = "1.29.2")
bazel_dep(name = "platforms", version = "0.0.5")
bazel_dep(name = "rules_oci", version = "0.3.8")
bazel_dep(name = "rules_pkg", version = "0.8.1")

oci = use_extension("@rules_oci//oci:extensions.bzl", "oci")
oci.pull(
    name = "distroless_cc",
    digest = "sha256:8aad707f96620ee89e27febef51b01c6ff244277a3560fcfcfbe68633ef09193",
    image = "gcr.io/distroless/cc",
    platforms = [
        "linux/amd64",
        "linux/arm64",
    ],
)
use_repo(oci, "distroless_cc")
```

Error message:
```
ERROR: An error occurred during the fetch of repository 'rules_oci~0.3.8~oci~distroless_cc_linux_arm64':
   Traceback (most recent call last):
        File "/private/var/tmp/_bazel_brian.myers/7c67ba1d63a5ecf4b1e2bbf1c99fe09a/external/rules_oci~0.3.8/oci/pull.bzl", line 333, column 13, in _oci_pull_impl
                fail("Unrecognized mediaType {} in manifest file".format(mf["mediaType"]))
Error in fail: Unrecognized mediaType application/vnd.oci.image.index.v1+json in manifest file
ERROR: <builtin>: fetching oci_pull_rule rule @rules_oci~0.3.8//oci:rules_oci~0.3.8~oci~distroless_cc_linux_arm64: Traceback (most recent call last):
        File "/private/var/tmp/_bazel_brian.myers/7c67ba1d63a5ecf4b1e2bbf1c99fe09a/external/rules_oci~0.3.8/oci/pull.bzl", line 333, column 13, in _oci_pull_impl
                fail("Unrecognized mediaType {} in manifest file".format(mf["mediaType"]))
Error in fail: Unrecognized mediaType application/vnd.oci.image.index.v1+json in manifest file
ERROR: /private/var/tmp/_bazel_brian.myers/7c67ba1d63a5ecf4b1e2bbf1c99fe09a/external/rules_oci~0.3.8~oci~distroless_cc/BUILD.bazel:1:6: @rules_oci~0.3.8~oci~distroless_cc//:distroless_cc depends on @rules_oci~0.3.8~oci~distroless_cc_linux_arm64//:distroless_cc_linux_arm64 in repository @rules_oci~0.3.8~oci~distroless_cc_linux_arm64 which failed to fetch. no such package '@rules_oci~0.3.8~oci~distroless_cc_linux_arm64//': Unrecognized mediaType application/vnd.oci.image.index.v1+json in manifest file
```

### CODE REVIEW

**Comment by thesayyn** [comment]
thank you for the PR. i have PR already fixing OCI mediatypes here https://github.com/bazel-contrib/rules_oci/pull/105/files.
I'd be happy to land the change for rules_pkg though.

**Comment by bcmyers** [comment]
Yah. Happy with whatever fixes the issue. @alexeagle asked me to add a rust example to https://github.com/aspect-build/bazel-examples and I hit this bug; so figured I'd send you a PR to fix it, but I'm indifferent to how it's fixed. Thanks!

File: oci/dependencies.bzl
Comment (alexeagle) [inline oci/dependencies.bzl:36]: OOPS

**Review by alexeagle — APPROVED** [review]

**Comment by alexeagle** [comment]
+1 to each PR doing just one thing, by the way - thanks

**Comment by thesayyn** [comment]
> Looking at your PR, though, it seems to do a lot more than fix this bug

right. most of them are already in their PRs. I need to rebase that PR after I land other PRs. If we are going to land this, let's use a method like in my pr. I don't think we need to keep a list of media types, we only got two for each.

**Comment by alexeagle** [comment]
I'm going to merge since it's minimal and green and to get Brian on our committer list - we can refactor to be less general if we like.

**Pull request merged**
